nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -209,24 +209,29 @@ impl SupportBundleCollector {
             SupportBundleState::Destroying => {
                 // Destroying is a terminal state; no one should be able to
                 // change this state from underneath us.
-                self.datastore.support_bundle_delete(
-                    opctx,
-                    &authz_bundle,
-                ).await.map_err(|err| {
-                    warn!(
-                        &opctx.log,
-                        "SupportBundleCollector: Could not delete 'destroying' bundle";
-                        "err" => %err
-                    );
-                    anyhow::anyhow!("Could not delete 'destroying' bundle: {:#}", err)
-                })?;
-
-                return Ok(
-                    DatabaseBundleCleanupResult::DestroyingBundleRemoved,
-                );
+                match self
+                    .datastore
+                    .support_bundle_delete(opctx, &authz_bundle)
+                    .await
+                {
+                    Ok(_) | Err(Error::NotFound { .. }) => {
+                        return Ok(DatabaseBundleCleanupResult::DestroyingBundleRemoved);
+                    }
+                    Err(err) => {
+                        warn!(
+                            &opctx.log,
+                            "SupportBundleCollector: Could not delete 'destroying' bundle";
+                            "err" => %err
+                        );
+                        anyhow::bail!(
+                            "Could not delete 'destroying' bundle: {:#}",
```

Comment:
And looking at the `Error` type here it doesn't seem like the variants encode the bundle id, but maybe it's easy enough to figure out from the dropshot request log?